pool/src/lib.rs

```diff
@@ -628,9 +648,12 @@ impl LendingPool {
             if user_config.is_using_as_collateral(env, i) {
                 let coll_coeff = Self::get_collateral_coeff(env, &curr_reserve)?;
 
-                let who_balance: i128 = mb_who_balance.unwrap_or_else(|| {
-                    STokenClient::new(env, &curr_reserve.s_token_address).balance(&who)
-                });
+                let who_balance: i128 =
+                    if mb_who_stoken_address == Some(curr_reserve.s_token_address.clone()) {
```

Comment:
This usually unpack as `if let Some(curr_reserve.s_token_address) = mb_who_stoken_address`

Reply:
My bad, it's can be written as now